libs/media-player/src/lib/video-prototype-player.tsx

```diff
@@ -0,0 +1,238 @@
+            // Video not yet loaded
+        }
+        if (isPlaying) {
+            videoRef.current!.pause();
+        } else {
+            videoRef.current!.play();
+        }
+        setIsPlaying(!isPlaying);
+    };
+
+    const seekInMedia = (time: number) => {
+        videoRef.current!.currentTime = time;
+    };
+
+    const seekInProgressBar = (event: React.MouseEvent<HTMLSpanElement>) => {
+        const percentProgressSelected =
+            (event.clientX - event.currentTarget.offsetLeft) / event.currentTarget.offsetWidth;
+
+        const newMediaTime = percentProgressSelected * videoRef.current!.duration;
+
+        seekInMedia(newMediaTime);
+
+        updateProgressBarAndMediaCurrentTime(newMediaTime);
+    };
+
+    const restartMedia = () => {
+        seekInMedia(0);
+
+        if (isPlaying) {
+            togglePlay();
+        }
+    };
+
+    const handlePlayProgress = () => {
+        const currentTime = videoRef.current!.currentTime;
+
+        updateProgressBarAndMediaCurrentTime(currentTime);
+    };
+
+    const updateProgressBarAndMediaCurrentTime = (time: number) => {
+        const duration = videoRef.current!.duration;
+
+        setMediaCurrentTimeForPresentation(time);
+
+        const progress = (time / duration) * 100;
+
+        setProgress(progress);
+    };
+
+    const markInPoint = () => {
+        setInPoint(videoRef.current!.currentTime);
+    };
+
+    const markOutPoint = () => {
+        setOutPoint(videoRef.current!.currentTime);
+    };
+
+    useEffect(() => {
+        if (isNullOrUndefined(videoUrl)) {
+            setVideoVerifiedState(VideoVerifiedState.isMissingAudioLink);
+            return;
+        }
+
+        const onCanPlay = () => {
+            setVideoVerifiedState(VideoVerifiedState.canPlay);
+        };
+
+        const onError = () => {
+            setVideoVerifiedState(VideoVerifiedState.error);
+        };
+
+        /**
+         * This video instance will never be played. We are using it to test whether
+         * the URL is valid.
+         */
+        const testVideo = document.createElement('video');
+
+        testVideo.src = videoUrl;
+
+        testVideo.addEventListener('canplaythrough', onCanPlay);
+
+        testVideo.addEventListener('error', onError);
+
+        videoRef.current!.addEventListener('progress', () => {
+            const bufferedVideo = videoRef.current!.buffered;
+
+            const bufferedVideoLength = bufferedVideo.length;
+
+            const bufferedEnd =
+                bufferedVideoLength > 0 ? bufferedVideo.end(bufferedVideoLength - 1) : 0;
+
+            const videoDuration = videoRef.current!.duration;
+
+            setBuffer((bufferedEnd / videoDuration) * 100);
+        });
+
+        document.addEventListener('keydown', (event) => {
+            const keys = ['i', 'o', 'j', 'k'];
+
+            const wasAnyKeyPressed = keys.some((key) => event.key === key);
+
+            if (wasAnyKeyPressed) {
```

Comment:
Can you add a cheat-sheet that says what the hotkeys are?

Reply:
Agreed, this was just a first take on something that needs a wider design conversation.